airbyte-integrations/bases/base-normalization/normalization/transform_catalog/utils.py

```diff
@@ -16,76 +17,125 @@ def remove_jinja(command: str) -> str:
     return str(command).replace("{{ ", "").replace(" }}", "")
 
 
+# transform oneOf nested list ot plain types list
+def get_plain_list_from_one_of_array(one_of_definition) -> list:
+    one_of_property = one_of_definition[data_type.ONE_OF_VAR_NAME]
+    plain_datatypes_list = list()
+    for prop in one_of_property:
+        plain_datatypes_list.append(prop[data_type.REF_TYPE_VAR_NAME])
+    return plain_datatypes_list
+
+
 def is_string(property_type) -> bool:
-    return property_type == "string" or "string" in property_type
+    if data_type.ONE_OF_VAR_NAME in property_type and data_type.ONE_OF_VAR_NAME in property_type:
+        plain_datatypes_list = get_plain_list_from_one_of_array(property_type)
+        return plain_datatypes_list == data_type.STRING_TYPE or data_type.STRING_TYPE in plain_datatypes_list
+    else:
+        return property_type == data_type.STRING_TYPE or data_type.STRING_TYPE in property_type
+
+
+def is_binary_datatype(property_type) -> bool:
```

Comment:
Sometimes this may a simple string. Sometimes dict, sometimes string :( 
![Selection_139](https://user-images.githubusercontent.com/4688354/206731628-adaaa156-1f42-43df-a7ca-1c23dc9f46a2.png)